AK/Bitmap.h

```diff
@@ -163,69 +157,151 @@ class Bitmap {
         return {};
     }
 
-    Optional<size_t> find_longest_range_of_unset_bits(size_t max_length, size_t& found_range_size) const
+    /**
+     * the function is supposed to find the next range of unset bits 
+     * starting from the @from value. 
+     * @from: the postition from which the search starts. The var
+     *        will be changed and new value is the offset of the
+     *        found block
+     * @min_length: minimum size of the range which will be returned
+     * @max_length: shows when to stop adding unset bits to the range.
+     *            This is used for optimisation, since the range of
+     *            unset bits can be long, and we don't need the whole
+     *            size, so we can stop when we've reached @max_length.
+     * It's recomended to use @min_length and @max_length, since they
+     * give a boost.
+     */
+    inline Optional<size_t> find_next_range_of_unset_bits(size_t& from, size_t min_length=1, size_t max_length=0xffffffff) const
     {
-        auto first_index = find_first_unset();
-        if (!first_index.has_value())
-            return {};
 
-        size_t free_region_start = first_index.value();
-        size_t free_region_size = 1;
+#ifdef AK_ARCH_I386
+        #define trailing_zeroes_cnt(x) (__builtin_ctz(x))
+#else
+        auto trailing_zeroes_cnt = [](u32 val) -> u32{
+            for (u8 i = 0; i < 32; ++i) {
+                if ((val >> i) & 1) {
+                    return i;
+                }
+            }
+            return 0;
+        };
+#endif /* AK_ARCH_I386 */
```

Comment:
Please put this in `AK/Platform.h`.
Also, please don't abbreviate random words. `trailing_zeroes_count` or `count_trailing_zeroes`.